vnext/Microsoft.ReactNative/Utils/ValueUtils.cpp

```diff
@@ -44,58 +47,105 @@ struct ColorComp {
   }
 };
 
-xaml::Media::Brush BrushFromColorObject(winrt::hstring resourceName) {
-  thread_local static std::map<winrt::hstring, winrt::weak_ref<xaml::Media::SolidColorBrush>> accentColorMap = {
-      {L"SystemAccentColor", {nullptr}},
-      {L"SystemAccentColorLight1", {nullptr}},
-      {L"SystemAccentColorLight2", {nullptr}},
-      {L"SystemAccentColorLight3", {nullptr}},
-      {L"SystemAccentColorDark1", {nullptr}},
-      {L"SystemAccentColorDark2", {nullptr}},
-      {L"SystemAccentColorDark3", {nullptr}},
-      {L"SystemListAccentLowColor", {nullptr}},
-      {L"SystemListAccentMediumColor", {nullptr}},
-      {L"SystemListAccentHighColor", {nullptr}}};
-
-  if (accentColorMap.find(resourceName) != accentColorMap.end()) {
-    if (auto brush = accentColorMap.at(resourceName).get()) {
-      return brush;
-    }
+xaml::Media::Brush BrushFromTheme(winrt::hstring resourceName) {
+  winrt::hstring xamlString =
+      L"<ResourceDictionary"
+      L"    xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'"
+      L"    xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'>"
+      L"  <SolidColorBrush"
+      L"      x:Key='" +
+      resourceName +
+      L"'"
+      L"      Color='{ThemeResource " +
+      resourceName +
+      "}' />"
+      L"</ResourceDictionary>";
+
+  auto dictionary{winrt::unbox_value<winrt::ResourceDictionary>(winrt::Markup::XamlReader::Load(xamlString))};
+
+  auto brush{winrt::unbox_value<xaml::Media::SolidColorBrush>(dictionary.Lookup(winrt::box_value(resourceName)))};
+  return brush;
+}
 
-    winrt::hstring xamlString =
-        L"<ResourceDictionary"
-        L"    xmlns='http://schemas.microsoft.com/winfx/2006/xaml/presentation'"
-        L"    xmlns:x='http://schemas.microsoft.com/winfx/2006/xaml'>"
-        L"  <SolidColorBrush"
-        L"      x:Key='" +
-        resourceName +
-        L"'"
-        L"      Color='{ThemeResource " +
-        resourceName +
-        "}' />"
-        L"</ResourceDictionary>";
+struct BrushCache : public std::map<winrt::hstring, winrt::weak_ref<xaml::Media::SolidColorBrush>> {
+  std::map<winrt::hstring, xaml::Media::Brush> m_map;
```

Comment:
I see we're inheriting from std::map, and separately embedding a map inside.

Do we want this object to be exposing all of the functionality of std::map?

Reply:
yeah good catch I had started with a subclass but then decided on encapsulating it instead.